### PR TITLE
More linting fixes

### PR DESCRIPTION
**Pull request by TGCrystal**
Fixes one of the checkboxes on #1088 
Also enforces some spacing around keywords and inside parenthesis

### CODE REVIEW

File: src/app/analog/models/eeobjects/sources/VoltageSource.ts [hunk 1]
```diff
@@ -15,30 +15,30 @@ import {TopBottomPositioner} from "analog/models/ports/positioners/TopBottomPosi
...
     infos: {
         ...VoltageInfo("v1", "Low Voltage",  0),
-        ...VoltageInfo( "V", "High Voltage", 5),
+        ...VoltageInfo("V", "High Voltage", 5),
            ...TimeInfo("td", "Delay Time",   0),
            ...TimeInfo("tr", "Rise Time",    0.01),
            ...TimeInfo("tf", "Fall Time",    0.01),
            ...TimeInfo("pw", "Pulse Width",  0.1),
-           ...TimeInfo( "p", "Period",       0.2),
+           ...TimeInfo("p", "Period",       0.2),
```
Comment (LeonMontealegre) [inline src/app/analog/models/eeobjects/sources/VoltageSource.ts:35]: The vertical align is messed up here

File: src/app/core/actions/selection/SelectAction.ts [hunk 2]
```diff
@@ -45,14 +45,11 @@ export class DeselectAction extends SelectAction {
...
+    return objs.reduce((acc, s) => acc.add(new SelectAction(selections, s)), new GroupAction([], "Select Action"));
 }
 
 export function CreateDeselectAllAction(selections: SelectionsWrapper): GroupAction {
     const objs = selections.get();
-    return objs.reduce((acc, s) => {
-        return acc.add(new DeselectAction(selections, s));
-    }, new GroupAction([], "Deselect All Action"));
+    return objs.reduce((acc, s) => acc.add(new DeselectAction(selections, s)),
+                                   new GroupAction([], "Deselect All Action"));
```
Comment (LeonMontealegre) [inline src/app/core/actions/selection/SelectAction.ts:54]: preferably:
```ts
return objs.reduce(
    (acc, s) => acc.add(new DeselectAction(selections, s)),
    new GroupAction([], "Deselect All Action")
);
```

File: src/app/core/rendering/CreateRenderers.ts [hunk 3]
```diff
@@ -33,7 +33,7 @@ export const CreateRenderers = <A, B1, B2, C1, C2, D, E>(
         toolRenderer: RendererInstance<D>;
         debugRenderer: RendererInstance<E>;
     }
-) => ({
+) => {return {
```
Comment (LeonMontealegre) [inline src/app/core/rendering/CreateRenderers.ts:36]: ?

File: src/app/core/rendering/renderers/IOLabelRenderer.ts [hunk 4]
```diff
@@ -35,7 +35,7 @@ export const IOLabelRenderer = (() => {
         const xBound = size.x/2 - IO_LABEL_DIR_PADDING - textWidth/2;
         const yBound = size.y/2 - 2*IO_LABEL_VERTICAL_TEXT_PADDING;
         const min = V(-xBound, -yBound);
-        const max = V( xBound,  yBound);
+        const max = V(xBound,  yBound);
```
Comment (LeonMontealegre) [inline src/app/core/rendering/renderers/IOLabelRenderer.ts:38]: vertical align :(

File: src/app/core/tools/handlers/PasteHandler.ts [hunk 5]
```diff
@@ -3,8 +3,7 @@ import {CopyPasteEvent, Event} from "core/utils/Events";
 import {EventHandler} from "../EventHandler";
 
 
-export const PasteHandler = (paste: (text: string) => boolean): EventHandler => {
-    return ({
+export const PasteHandler = (paste: (text: string) => boolean): EventHandler => ({
```
Comment (LeonMontealegre) [inline src/app/core/tools/handlers/PasteHandler.ts:6]: need to de-indent the following lines

File: src/app/core/tools/handlers/PasteHandler.ts [hunk 6]
```diff
@@ -13,5 +12,4 @@ export const PasteHandler = (paste: (text: string) => boolean): EventHandler =>
         getResponse: (_, { ev }: CopyPasteEvent) => {
             paste(ev.clipboardData!.getData("text/plain"));
         },
-    });
-}
+    })
```
Comment (LeonMontealegre) [inline src/app/core/tools/handlers/PasteHandler.ts:15]: semi-colon at end

File: src/app/core/tools/handlers/SaveHandler.ts [hunk 7]
```diff
@@ -4,13 +4,11 @@ import {Event}       from "core/utils/Events";
 import {EventHandler} from "../EventHandler";
 
 
-export const SaveHandler = (save: () => void): EventHandler => {
-    return ({
+export const SaveHandler = (save: () => void): EventHandler => ({
```
Comment (LeonMontealegre) [inline src/app/core/tools/handlers/SaveHandler.ts:7]: need to de-indent the following lines

File: src/app/core/utils/Input.ts [hunk 8]
```diff
@@ -91,11 +91,9 @@ export class Input {
...
         // Check if some combination has every key pressed and newKey is one of them
         //  and return true if that's the case
-        return PREVENTED_COMBINATIONS.some((combination) => {
-            return combination.flat().includes(newKey) &&
-                   combination.every(keys =>
-                       keys.some(key => this.isKeyDown(key)));
-        });
+        return PREVENTED_COMBINATIONS.some((combination) => combination.flat().includes(newKey) &&
+                   combination.every((keys) =>
+                       keys.some((key) => this.isKeyDown(key))));
```
Comment (LeonMontealegre) [inline src/app/core/utils/Input.ts:96]: spacing is weird here

File: src/app/core/utils/math/Rect.ts [hunk 9]
```diff
@@ -135,7 +135,7 @@ export class Rect {
             Rect.From({ left: rect.right, right: this.right, top: this.top,    bottom: rect.top    }),
             Rect.From({ left: rect.right, right: this.right, top: rect.top,    bottom: rect.bottom }),
             Rect.From({ left: rect.right, right: this.right, top: rect.bottom, bottom: this.bottom }),
-        ].filter(r => r.width > 0 && r.height > 0);
+        ].filter((r) => r.width > 0 && r.height > 0);
```
Comment (LeonMontealegre) [inline src/app/core/utils/math/Rect.ts:138]: ,parenthesis around the filter body, i think the rule of thumb is that if you have a statement that contains a space, i.e. a comparision like ` == ` or ` > ` or ` && ` or something, there should be parenthesis

File: src/app/digital/actions/addition/BusActionFactory.ts [hunk 10]
```diff
@@ -36,40 +36,38 @@ export function CreateBusAction(outputPorts: OutputPort[], inputPorts: InputPort
...
     // Get angles of each port in local space
-    const outputAngles = outputTargetPositions.map(p => p.angle());
-    const inputAngles  =  inputTargetPositions.map(p => p.angle()).map(a => (a < 0 ? a + 2*Math.PI : a));
+    const outputAngles = outputTargetPositions.map((p) => p.angle());
+    const inputAngles  =  inputTargetPositions.map((p) => p.angle()).map((a) => (a < 0 ? a + 2*Math.PI : a));
 
     // Associate the ports with their angle
     const outputMap = new Map(outputAngles.map((a, i) => [a, outputPorts[i]]));
-    const inputMap  = new Map( inputAngles.map((a, i) => [a,  inputPorts[i]]));
+    const inputMap  = new Map(inputAngles.map((a, i) => [a,  inputPorts[i]]));
```
Comment (LeonMontealegre) [inline src/app/digital/actions/addition/BusActionFactory.ts:70]: vertical align :(

File: src/app/digital/models/ioobjects/other/ICData.ts [hunk 11]
```diff
@@ -171,9 +171,9 @@ export class ICData {
         //  things like ConstantHigh and ConstantLow which aren't interactive
         const INPUT_WHITELIST = [Switch, Button];
         const OUTPUT_WHITELIST = [LED];
-        const inputs  = copies.getInputs().filter( i => INPUT_WHITELIST.some( (type) => i instanceof type));
-        const outputs = copies.getOutputs().filter(o => OUTPUT_WHITELIST.some((type) => o instanceof type));
-        const others  = copies.getComponents().filter(c => (!inputs.includes(c) && !outputs.includes(c)));
+        const inputs  = copies.getInputs().filter((i) => INPUT_WHITELIST.some((type) => i instanceof type));
+        const outputs = copies.getOutputs().filter((o) => OUTPUT_WHITELIST.some((type) => o instanceof type));
+        const others  = copies.getComponents().filter((c) => (!inputs.includes(c) && !outputs.includes(c)));
```
Comment (LeonMontealegre) [inline src/app/digital/models/ioobjects/other/ICData.ts:176]: vertical align :(

File: src/app/digital/utils/ComponentUtils.ts [hunk 12]
```diff
@@ -76,13 +76,11 @@ export function LazyConnect(source: DigitalComponent, destination: DigitalCompon
  * @returns          True if the ICData is being used somewhere, false otherwise.
  */
 export function IsICDataInUse(designer: DigitalCircuitDesigner, data: ICData): boolean {
-    const checkInUse = (objs: IOObject[]): boolean => {
-        return objs.some(o =>
+    const checkInUse = (objs: IOObject[]): boolean => objs.some((o) =>
```
Comment (LeonMontealegre) [inline src/app/digital/utils/ComponentUtils.ts:79]: spacing is weird here

File: src/app/digital/utils/DigitalVersionConflictResolver.ts [hunk 13]
```diff
@@ -85,12 +85,10 @@ export function VersionConflictResolver(fileContents: string | Circuit): string
...
-                data: transformation.reduce((props, { prevKey, newKey, defaultVal }) => {
-                    return {
-                        ...props,
-                        [newKey]: (data[prevKey] ?? defaultVal),
-                    };
-                }, {}),
+                data: Object.fromEntries(transformation.map(({ prevKey,
+                                                               newKey,
+                                                               defaultVal,
+                                                            }) => [newKey, (data[prevKey] ?? defaultVal)])),
```
Comment (LeonMontealegre) [inline src/app/digital/utils/DigitalVersionConflictResolver.ts:91]: spacing is weird

File: src/app/tests/digital/utils/actions/BusAction.test.ts [hunk 14]
```diff
@@ -77,7 +77,7 @@ describe("Bus Action", () => {
         const [i1, o1] = Place(new ConstantNumber(), new BCDDisplay());
 
         i1.setPos(V(-100, 0));
-        o1.setPos(V( 100, 0));
+        o1.setPos(V(100, 0));
```
Comment (LeonMontealegre) [inline src/app/tests/digital/utils/actions/BusAction.test.ts:80]: vertical align :( (on entire file)

File: src/app/tests/digital/utils/math/Matrix.test.ts [hunk 15]
```diff
@@ -68,10 +68,10 @@ describe("Matrix2x3", () => {
                 m.rotate(Math.PI / 2);
                                         // [0 -1 0]
                                         // [1  0 0]
-                expect(m["mat"][0]).toBeCloseTo( 0, 1e-3);
-                expect(m["mat"][1]).toBeCloseTo( 1, 1e-3);
+                expect(m["mat"][0]).toBeCloseTo(0, 1e-3);
```
Comment (LeonMontealegre) [inline src/app/tests/digital/utils/math/Matrix.test.ts:71]: vertical align :(

File: src/app/tests/digital/utils/math/Transform.test.ts [hunk 16]
```diff
@@ -193,17 +193,17 @@ describe("Transform", () => {
             { // Translated
                 const t = new Transform(V(5, -5), V(1, 1), 0);
                 expect(t.toLocalSpace(V(0,  0))).toApproximatelyEqual(V(-5, 5));
-                expect(t.toLocalSpace(V(5, -5))).toApproximatelyEqual(V( 0, 0));
+                expect(t.toLocalSpace(V(5, -5))).toApproximatelyEqual(V(0, 0));
```
Comment (LeonMontealegre) [inline src/app/tests/digital/utils/math/Transform.test.ts:196]: vertical align :(

File: src/site/pages/analog/src/containers/AnalogItemNav/index.tsx [hunk 17]
```diff
@@ -21,8 +21,8 @@ export const AnalogItemNav = ({ info }: Props) => {
             throw new Error(`AnalogItemNav: Can't find ID for component ${c.getName()}`);
 
         // Get path within config of ItemNav icon
-        const section = itemNavConfig.sections.find(s => s.items.find(i => i.id === id));
-        const item = section?.items.find(i => i.id === id);
+        const section = itemNavConfig.sections.find((s) => s.items.find((i) => i.id === id));
+        const item = section?.items.find((i) => i.id === id);
```
Comment (LeonMontealegre) [inline src/site/pages/analog/src/containers/AnalogItemNav/index.tsx:25]: parenthesis on inner .find body

File: src/site/pages/analog/src/utils/hooks/useAnalog.ts [hunk 18]
```diff
@@ -6,10 +6,7 @@ import {AppState} from "site/analog/state";
 import {AllActions} from "site/analog/state/actions";
 
 
-export const useAnalogDispatch = () => {
-    return useDispatch<ThunkDispatch<AppState, undefined, AllActions>>();
-}
+export const useAnalogDispatch = () => useDispatch<ThunkDispatch<AppState, undefined, AllActions>>()
```
Comment (LeonMontealegre) [inline src/site/pages/analog/src/utils/hooks/useAnalog.ts:9]: semi-colon at end

File: src/site/pages/digital/src/containers/DigitalItemNav/index.tsx [hunk 19]
```diff
@@ -158,8 +158,8 @@ export const DigitalItemNav = ({ info }: Props) => {
             throw new Error(`DigitalItemNav: Can't find ID for component ${c.getName()}`);
 
         // Get path within config of ItemNav icon
-        const section = config.sections.find(s => s.items.find(i => i.id === id));
-        const item = section?.items.find(i => i.id === id);
+        const section = config.sections.find((s) => s.items.find((i) => i.id === id));
+        const item = section?.items.find((i) => i.id === id);
```
Comment (LeonMontealegre) [inline src/site/pages/digital/src/containers/DigitalItemNav/index.tsx:162]: parenthesis in inside body

File: src/site/pages/digital/src/utils/hooks/useDigital.ts [hunk 20]
```diff
@@ -6,13 +6,9 @@ import {AppState} from "site/digital/state";
 import {AllActions} from "site/digital/state/actions";
 
 
-export const useDigitalDispatch = () => {
-    return useDispatch<ThunkDispatch<AppState, undefined, AllActions>>();
-}
+export const useDigitalDispatch = () => useDispatch<ThunkDispatch<AppState, undefined, AllActions>>()
```
Comment (LeonMontealegre) [inline src/site/pages/digital/src/utils/hooks/useDigital.ts:9]: semicolon at end

File: src/site/shared/containers/HistoryBox/index.tsx [hunk 21]
```diff
@@ -82,9 +82,9 @@ const GroupActionEntry = ({ g, isRedo }: GroupActionEntryProps) => {
             {displayExtraInfo && g.getCustomInfo?.()?.map((obj, i) =>
                 <div key={`group-action-extrainfo-${i}`} className="historybox__groupentry__extrainfo">{obj}</div>
             )}
-            {!isCollapsed && g.getActions().map((a, i) => {
-                return (<HistoryEntry key={`group-action-entry-${i}`} a={a} isRedo={isRedo} />);
-            })}
+            {!isCollapsed && g.getActions().map((a, i) => (<HistoryEntry key={`group-action-entry-${i}`}
+                                                                         a={a}
+                                                                         isRedo={isRedo} />))}
```
Comment (LeonMontealegre) [inline src/site/shared/containers/HistoryBox/index.tsx:87]: spacing is weird

File: src/site/shared/containers/ItemNav/index.tsx [hunk 22]
```diff
@@ -201,16 +201,14 @@ export const ItemNav = <D,>({ info, config, additionalData, getImgSrc, onDelete,
         const H_MARGIN = 30, ITEM_WIDTH = 100;
 
         const numPerSection = Math.floor((w - H_MARGIN) / ITEM_WIDTH);
-        return config.sections.reduce((prev, section) => {
-            return [
+        return config.sections.reduce((prev, section) => [
```
Comment (LeonMontealegre) [inline src/site/shared/containers/ItemNav/index.tsx:204]: need to de-indent body

File: src/site/shared/utils/hooks/useShared.ts [hunk 23]
```diff
@@ -6,13 +6,9 @@ import {SharedAppState} from "shared/state";
 import {AllSharedActions} from "shared/state/actions";
 
 
-export const useSharedDispatch = () => {
-    return useDispatch<ThunkDispatch<SharedAppState, undefined, AllSharedActions>>();
-}
+export const useSharedDispatch = () => useDispatch<ThunkDispatch<SharedAppState, undefined, AllSharedActions>>()
```
Comment (LeonMontealegre) [inline src/site/shared/utils/hooks/useShared.ts:9]: semi-colon

**Review by LeonMontealegre — COMMENTED** [review]
A lot of the problems are with vertical align, but idk how we're gonna fix that.
Would almost certainly require a custom plugin that allows spaces in parenthesis if it detects some sort of vertical alignment but detecting that sounds complicated but im not entirely sure what to do in the meantime

we'll have to discuss further

File: src/app/core/utils/Input.ts (hunk 8 above)
Comment (TGCrystal) [reply src/app/core/utils/Input.ts:96]: What spacing do you want here?

Comment (LeonMontealegre) [reply src/app/core/utils/Input.ts:96]: probably something like:
```ts
return PREVENTED_COMBINATIONS.some((combination) => (
    combination.flat().includes(newKey) &&
    combination.every((keys) => (
        keys.some((key) => this.isKeyDown(key))
    )
));
```

File: src/app/digital/rendering/ioobjects/outputs/LEDRenderer.ts [hunk 24]
```diff
@@ -17,34 +17,28 @@ import {Circle} from "core/rendering/shapes/Circle";
...
-                gradient.addColorStop(0.4152, `rgba(${col.r}, ${col.g}, ${col.b}, ${LED_LIGHT_INTENSITY})`);
-                gradient.addColorStop(1, `rgba(${col.r}, ${col.g}, ${col.b}, 0)`);
-
-                // Draw circle w/ gradient as fill
-                renderer.draw(new Circle(V(), LED_LIGHT_RADIUS), new Style(gradient));
-            }
-        },
-    };
-})();
+
```
Comment (LeonMontealegre) [inline src/app/digital/rendering/ioobjects/outputs/LEDRenderer.ts:20]: Where'd the comment go?

File: src/app/digital/utils/ComponentUtils.ts (hunk 12 above)
Comment (LeonMontealegre) [reply src/app/digital/utils/ComponentUtils.ts:79]: should be like:
```ts
const checkInUse = (objs: IOObject[]): boolean => objs.some((o) => (
    o instanceof IC &&
    (o.getData() === data ||
        // Recursively check if this IC depends on the given data
        checkInUse(o.getData().getGroup().toList()))
));
```

File: src/app/core/actions/selection/SelectAction.ts [hunk 25]
```diff
@@ -45,14 +45,12 @@ export class DeselectAction extends SelectAction {
...
 }
 
 export function CreateDeselectAllAction(selections: SelectionsWrapper): GroupAction {
     const objs = selections.get();
-    return objs.reduce((acc, s) => {
-        return acc.add(new DeselectAction(selections, s));
-    }, new GroupAction([], "Deselect All Action"));
+    return objs.reduce(
+        (acc, s) => acc.add(new DeselectAction(selections, s)),
+        new GroupAction([], "Deselect All Action"));
```
Comment (LeonMontealegre) [inline src/app/core/actions/selection/SelectAction.ts:55]: put the last `);` on the next line

File: src/app/digital/utils/DigitalVersionConflictResolver.ts [hunk 26]
```diff
@@ -85,12 +85,8 @@ export function VersionConflictResolver(fileContents: string | Circuit): string
...
             data["props"] = {
                 type: "",
-                data: transformation.reduce((props, { prevKey, newKey, defaultVal }) => {
-                    return {
-                        ...props,
-                        [newKey]: (data[prevKey] ?? defaultVal),
-                    };
-                }, {}),
+                data: Object.fromEntries(transformation.map(({ prevKey, newKey, defaultVal }) =>
+                    [newKey, (data[prevKey] ?? defaultVal)])),
```
Comment (LeonMontealegre) [inline src/app/digital/utils/DigitalVersionConflictResolver.ts:89]: should be:
```ts
data: Object.fromEntries(
    transformation.map(({ prevKey, newKey, defaultVal }) => 
        [newKey, (data[prevKey] ?? defaultVal)]
    )
)

File: src/site/shared/containers/HistoryBox/index.tsx [hunk 27]
```diff
@@ -82,9 +82,10 @@ const GroupActionEntry = ({ g, isRedo }: GroupActionEntryProps) => {
             {displayExtraInfo && g.getCustomInfo?.()?.map((obj, i) =>
                 <div key={`group-action-extrainfo-${i}`} className="historybox__groupentry__extrainfo">{obj}</div>
             )}
-            {!isCollapsed && g.getActions().map((a, i) => {
-                return (<HistoryEntry key={`group-action-entry-${i}`} a={a} isRedo={isRedo} />);
-            })}
+            {!isCollapsed && g.getActions().map((a, i) =>
+                (<HistoryEntry key={`group-action-entry-${i}`}
+                               a={a}
+                               isRedo={isRedo} />))}
```
Comment (LeonMontealegre) [inline src/site/shared/containers/HistoryBox/index.tsx:88]: i think something more like:
```ts
{!isCollapsed && g.getActions().map((a, i) => (
    <HistoryEntry key={`group-action-entry-${i}`}
                  a={a}
                  isRedo={isRedo} />
))}
```

**Review by LeonMontealegre — CHANGES_REQUESTED** [review]
only a few more spacing issues

**Review by LeonMontealegre — APPROVED** [review]

**Review by LeonMontealegre — APPROVED** [review]

**Pull request merged**